vsphere/virtual_machine_config_structure.go

```diff
@@ -0,0 +1,664 @@
+// need to work with a VirtualMachineConfigSpec.
+func schemaVirtualMachineConfigSpec() map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
+		// VirtualMachineBootOptions
+		"boot_delay": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "The number of milliseconds to wait before starting the boot sequence.",
+		},
+		"efi_secure_boot_enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "When the boot type set in firmware is efi, this enables EFI secure boot.",
+		},
+		"boot_retry_delay": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Default:     10000,
+			Description: "The number of milliseconds to wait before retrying the boot sequence. This only valid if boot_retry_enabled is true.",
+		},
+		"boot_retry_enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "If set to true, a virtual machine that fails to boot will try again after the delay defined in boot_retry_delay.",
+		},
+
+		// VirtualMachineFlagInfo
+		"enable_disk_uuid": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Expose the UUIDs of attached virtual disks to the virtual machine, allowing access to them in the guest.",
+		},
+		"hv_mode": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      string(types.VirtualMachineFlagInfoVirtualExecUsageHvAuto),
+			Description:  "The (non-nested) hardware virtualization setting for this virtual machine. Can be one of hvAuto, hvOn, or hvOff.",
+			ValidateFunc: validation.StringInSlice(virtualMachineVirtualExecUsageAllowedValues, false),
+		},
+		"ept_rvi_mode": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      string(types.VirtualMachineFlagInfoVirtualMmuUsageAutomatic),
+			Description:  "The EPT/RVI (hardware memory virtualization) setting for this virtual machine. Can be one of automatic, on, or off.",
+			ValidateFunc: validation.StringInSlice(virtualMachineVirtualMmuUsageAllowedValues, false),
+		},
+		"enable_logging": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Enable logging on this virtual machine.",
+		},
+
+		// ToolsConfigInfo
+		"sync_time_with_host": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable guest clock synchronization with the host. Requires VMware tools to be installed.",
+		},
+		"run_tools_scripts_after_power_on": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Enable the execution of post-power-on scripts when VMware tools is installed.",
+		},
+		"run_tools_scripts_after_resume": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Enable the execution of post-resume scripts when VMware tools is installed.",
+		},
+		"run_tools_scripts_before_guest_reboot": {
+			Type:        schema.TypeBool,
+			Optional:    true,
```

Comment:
does this want a default value to match the others in this block?

Reply:
The default here is false - it was true before, but I think something to do with VM configuration forces this to `false` (maybe due to guest type)? This was causing spurious diffs, so the default was moved to `false`. I'll make a TODO to look after it.

Reply:
Opened as #254